Userland/Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -803,6 +804,33 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::match_all)
     return TRY(Value(rx).invoke(global_object, *vm.well_known_symbol_match_all(), js_string(vm, move(string))));
 }
 
+// 22.1.3.14 String.prototype.normalize ( [ form ] ), https://tc39.es/ecma262/#sec-string.prototype.normalize
+JS_DEFINE_NATIVE_FUNCTION(StringPrototype::normalize)
+{
+    // 1. Let O be ? RequireObjectCoercible(this value).
+    auto this_object = TRY(require_object_coercible(global_object, vm.this_value(global_object)));
+
+    // 2. Let S be ? ToString(O).
+    auto string = TRY(this_object.to_utf16_string(global_object));
```

Comment:
We have `ak_string_from` and `utf16_string_from` at the top of this file to abstract these steps.

If `string` doesn't need to be UTF-16, prefer to use `ak_string_from` to create a UTF-8 string. Generally, we should only transcode to UTF-16 if we need to know UTF-16 specific things, e.g. the number of 16-bit code units in the string. Otherwise we prefer to stick with UTF-8.